okhttp/src/main/java/okhttp3/Headers.kt

```diff
@@ -122,6 +130,10 @@ class Headers private constructor(
     return result
   }
 
+  override operator fun iterator(): Iterator<Pair<String, String>> {
+    return Array(size) { name(it) to value(it) }.iterator()
```

Comment:
This is unfortunately heavy on allocation

Reply:
yeah, and as a consequence I’m not using it internally as much as I’d prefer otherwise.

I think the alternative is an inline function, `forEachHeader()`, but I don’t think that’s as flexible.

Reply:
One option this pushes us towards is changing the internal store from alternating names and values to an array of pairs. That could be fewer allocations if you iterate repeatedly.